pkg/controller/permissions/permissions_check.go

```diff
@@ -67,6 +72,40 @@ func CheckImagePermissions(req router.Request, resp router.Response) error {
 			details.AppImage.Digest, appImage.Digest)
 	}
 
+	// If enabled, check if the Acorn images are authorized to request the defined permissions.
+	if enabled, err := config.GetFeature(req.Ctx, req.Client, profiles.FeatureImageRoleAuthorizations); err != nil {
+		return err
+	} else if enabled {
+		imageName := appImage.Name
+
+		// E.g. for child Acorns, the appImage.Name is the image ID, but we need the original image name (with registry/repo)
+		// to check for the signatures
+		if oi, ok := app.GetAnnotations()[labels.AcornOriginalImage]; ok {
+			imageName = oi
+		}
```

Comment:
Just realized that this may be bad, since we're looking at the staged appImage which may or may not match that annotation - does anyone have another idea?

Reply:
Figured out that this is actually OK, since the label is always up-to-date with the staged appImage :heavy_check_mark:

Reply:
Awesome! I couldn't think of any other ideas for this so I am glad that this will work